tests/rest/client/test_account.py

```diff
@@ -794,21 +794,21 @@ def test_add_valid_email_second_time_canonicalise(self) -> None:
     def test_add_email_no_at(self) -> None:
         self._request_token_invalid_email(
             "address-without-at.bar",
-            expected_errcode=Codes.BAD_JSON,
+            expected_errcode=Codes.INVALID_PARAM,
             expected_error="Unable to parse email address",
         )
```

Comment:
This looks like a nice improvement, and presumably corresponds to the changes in `validate_json_object`. Do we know why this gave BAD_JSON before? (i.e. what is the type of exception caught by `validate_json_object` here?)

Reply:
My hope is that I didn't break any valid assumptions with this, but looking at the tests it seemed correct.